### PR TITLE
page:comments ft. edit post delete post loader add

### PR DESCRIPTION
**Pull request by manojsarna**
- [x] Image Upload As Post Added
- [x] Comments Page Added
- [x] Profile Posts Added
- [x] Edit Profile Added
- [x] Follow UnFollow Added
- [x] Functionality Added - Edit Post , Delete Post , Loaders Added

###Check these Files 

[/CommentCard.jsx](https://github.com/manojsarna/manoj-smasher-socialmedia-react/compare/post-comments-edit-post-functionality?expand=1#diff-33f3c0965b844ec18736b42af1bdb984cd2cc3ed88d72a0887bee73f6c97a5c0)
[ditPostModal.jsx](https://github.com/manojsarna/manoj-smasher-socialmedia-react/compare/post-comments-edit-post-functionality?expand=1#diff-fc19a2e4fd36ed271577f8bcdc422cb3be7d1bb93ec217c56729fcb757d66276)
[Home.jsx](https://github.com/manojsarna/manoj-smasher-socialmedia-react/compare/post-comments-edit-post-functionality?expand=1#diff-172462f7215de52769ba39af6ddb83fcc09fcaefa36a75651b75c68a0e43a05a)

### Ignore Rest of the Files

image
image

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -38,7 +46,7 @@ function App() {
         <Route element={<PrivateRoute />}>
           <Route path="/home" element={<Home />} />
           <Route path="/:username" element={<User />} />
-          <Route path="/:username/:postId" element={<Post />} />
+          <Route path="/:username/:postId" element={<PostComments />} />
```
Comment (iRohitGaur) [inline src/App.jsx:49]: suggestion: name of the component doesnt explain what the content will be. Please put a better name

File: src/components/comment-card/CommentCard.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const { width } = useWindowDimensions();
+
+  return (
+    <div className="sm-single-comment-container">
+      <Link to={`/${comment.username}`} className="sm-home-avatar">
+        <div title="Go To Profile">
+          <div className="avatar avatar-hover s-s ">
+            <img src={`${comment.profilePhoto}`} alt="badminton" />
+          </div>
+        </div>
```
Comment (iRohitGaur) [inline src/components/comment-card/CommentCard.jsx:20]: suggestion: you can use less div here

File: src/components/post/editpostmodal.css [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  align-items: center;
+  padding: 2rem;
+}
+
+.sm-modal-playilst-wrapper {
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  gap: 1rem;
+  overflow: hidden;
```
Comment (iRohitGaur) [inline src/components/post/editpostmodal.css:95]: suggestion: check this once, might not be needed

File: src/components/post/post.css [hunk 4]
```diff
@@ -73,6 +92,26 @@
...
+  border-radius: 10px;
+}
+
+.sm-post-content-image-container img {
+  border-radius: 10px;
+}
+
 @media only screen and (max-width: 1024px) {
   /* .sm-post-cta-container {
   } */
```
Comment (iRohitGaur) [inline src/components/post/post.css:117]: suggestion: remove commented code

File: src/hooks/getHomeFeed.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export const getHomeFeed = (posts, user) => {
+  if (posts.length === 0) return posts;
+  return posts.filter((post) => user.following.username === post.username);
+};
+// let tempPosts = posts.filter((item) =>
+// user.following.find((tempUser) => tempUser.username === item.username)
+// ).concat();
```
Comment (iRohitGaur) [inline src/hooks/getHomeFeed.js:7]: suggestion: remove commented code

File: src/pages/home/Home.jsx [hunk 6]
```diff
@@ -1,34 +1,69 @@
...
+  });
+
   const [selectSortValue, setSelectSortValue] = useState("Latest");
+  const loading = useSelector((state) => state.posts.loading);
+
+  const [isLoading, setIsLoading] = useState(false);
   const user = useSelector((state) => state.auth.user);
   const dispatch = useDispatch();
   const posts = useSelector((state) => state.posts.posts);
   let sortBy = useSelector((state) => state.posts.sortBy);
```
Comment (iRohitGaur) [inline src/pages/home/Home.jsx:32]: suggestion: you can destructure `posts` and `sortBy` from `state.posts` directly

File: src/pages/home/Home.jsx [hunk 7]
```diff
@@ -1,34 +1,69 @@
...
+    await fetch(process.env.REACT_APP_CLOUDINARY_API_URL, requestOptions)
+      .then((response) => response.json())
+      .then((json) => {
+        dispatch(
+          createPost({ content: postDetails.content, postImage: json.url })
+        );
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error(error);
```
Comment (iRohitGaur) [inline src/pages/home/Home.jsx]: suggestion: remove `console.log`

File: src/pages/home/Home.jsx [hunk 8]
```diff
@@ -1,34 +1,69 @@
...
 
   const postHandler = (postDetails) => {
-    if (postDetails !== "") {
-      if (postDetails.length > 5) {
-        dispatch(createPost(postDetails));
-        setPostDetails("");
+    if (postDetails.content !== "") {
+      if (postDetails.content.length > 5) {
+        if (postDetails.postImage !== "") {
+          cloudinaryPost(postDetails);
```
Comment (iRohitGaur) [inline src/pages/home/Home.jsx:63]: suggestion: you can handle this with `&&`

File: src/pages/post-comments/PostComments.jsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import { InfinitySpin as Loader } from "react-loader-spinner";
+import "./postcomments.css";
+import { toast } from "react-toastify";
+
+export function PostComments() {
+  const dispatch = useDispatch();
+  const { postId } = useParams();
+  const allPosts = useSelector((state) => state.posts.posts);
+  const currentPost = allPosts.find((item) => item._id === postId);
+  const loading = useSelector((state) => state.posts.loading);
```
Comment (iRohitGaur) [inline src/pages/post-comments/PostComments.jsx:15]: suggestion: you can destructure `allPosts` and `loading` directly from `state.posts`

File: src/pages/user/User.jsx [hunk 10]
```diff
@@ -1,35 +1,137 @@
...
+                <button
+                  className="btn btn-primary btn-bolder btn-round"
+                  title={followingThisUser ? "Unfollow" : "Follow"}
+                  // onClick={() => {
+                  //   if (followingThisUser) {
+                  //     dispatch(unfollowUser(profileUser._id));
+                  //   } else {
+                  //     dispatch(followUser(profileUser._id));
+                  //   }
+                  // }}
```
Comment (iRohitGaur) [inline src/pages/user/User.jsx]: suggestion: remove commented code

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions, please look into it.

Good to merge with updates 👍🏻

**Review by b0llu — APPROVED** [review]
Very Wow! You can merge this PR. Great work manoj.

File: src/App.jsx (hunk 1 above)
Comment (tarunsankhla) [inline src/App.jsx:49]: Don't you string directly create variable

File: src/components/comment-card/CommentCard.jsx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+      </Link>
+      <div className="sm-single-post-content">
+        <div className="sm-post-user-info-date">
+          <Link
+            to={`/${comment.username}`}
+            className="sm-remove-link-props"
+            title="Go To Profile"
+          >
+            <span className="sm-post-name">{`${comment.firstName} ${comment.lastName} `}</span>
+            <span className="sm-post-username">{`  @${comment.username} `}</span>
```
Comment (tarunsankhla) [inline src/components/comment-card/CommentCard.jsx:30]: Suggestion use semantic html here you could have used paragraph tag

File: src/components/post/Post.jsx [hunk 12]
```diff
@@ -1,23 +1,28 @@
...
+import { MdDelete } from "react-icons/md/";
 
 import { formatNumber } from "../../hooks/formatNumber";
 import {
   addToBookmarks,
+  deletePost,
   dislikePost,
   likePost,
   removeFromBookmarks,
 } from "../../redux/reducers/postsSlice";
```
Comment (tarunsankhla) [inline src/components/post/Post.jsx:22]: Use absolute path

File: src/components/post/editpostmodal.css [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+.sm-modal-content .sm-card-dismiss {
+  border: 0;
+  color: var(--COLOR-HEADER-FONT-DARK);
+  position: absolute;
+  top: 1px;
+  right: 1px;
+  font-size: 2.4rem;
+  width: fit-content;
+  border-radius: 0;
```
Comment (tarunsankhla) [inline src/components/post/editpostmodal.css:31]: Border radius is default 0 them why explicitly reassign

File: src/hooks/sortPosts.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    case "Oldest": {
+      finalPosts = finalPosts.sort(
+        (a, b) => new Date(a.createdAt) - new Date(b.createdAt)
+      );
+      break;
+    }
+    default:
+      break;
+  }
+  return finalPosts;
```
Comment (tarunsankhla) [inline src/hooks/sortPosts.js:25]: Return this in default  instead of just break

File: src/pages/home/home.css [hunk 15]
```diff
@@ -82,6 +82,25 @@
...
+  position: absolute;
+  top: 2px;
+  right: 6px;
+  color: var(--COLOR-HEADER-FONT-DARK);
+  font-size: 4rem;
+}
+
 /* MEDIA QUERIES */
 
 @media only screen and (max-width: 768px) {
```
Comment (tarunsankhla) [inline src/pages/home/home.css:106]: Use const for media queries width and height

File: src/pages/post-comments/PostComments.jsx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+              placeholder="Enter Comment..."
+              name="valid"
+              value={comment}
+              onChange={(e) => setComment(e.target.value)}
+            />
+
+            <button
+              className="sm-comment-btn"
+              onClick={() => {
+                if (comment !== "") {
```
Comment (tarunsankhla) [inline src/pages/post-comments/PostComments.jsx:43]: Move this into a method

File: src/pages/user/User.jsx [hunk 17]
```diff
@@ -1,35 +1,137 @@
...
+                <button
+                  className="sm-category-outline-btn sm-active"
+                  title="Edit Profile"
+                >
+                  <i className="fas fa-edit"></i>
+                </button>
+                <button
+                  className="sm-category-outline-btn sm-active"
+                  title="Logout Now"
+                  onClick={() => dispatch(logout())}
```
Comment (tarunsankhla) [inline src/pages/user/User.jsx:89]: Instead of callback directly use  logout

**Review by tarunsankhla — APPROVED** [review]
Overall great just minor changes

**Pull request merged**